bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/ValueUtils.java

```diff
@@ -241,4 +241,19 @@ public static BTypedesc getTypedescValue(Type type, BValue value) {
         return new TypedescValueImpl(type);
     }
 
+    /**
+     * Provide the Typedesc Value depending on the immutability of a value.
+     * @param isReadOnly Indicates that the type is a read-only
+     * @param value Ballerina value
+     * @param typedescValue typedesc value
+     * @return     typedesc with the suitable type
+     */
+    public static BTypedesc getTypedescValue(Boolean isReadOnly, BValue value, TypedescValueImpl typedescValue) {
```

Comment:
Isn't `typdescValue` here indicating the inherent type used when constructing the list? Shall we rename the param to something more reflective of what it is?